### PR TITLE
travis: test zephyr script and save built binary

### PR DESCRIPTION
**Pull request by mateusz-holenko**
This adds testing of `build-zephyr.sh` script on travis CI and saving the generated binary to prebuilt repo.

### CODE REVIEW

**Comment by mithro** [comment]
Can you add the Travis config to run the Zephyr build?

**Comment by mateusz-holenko** [comment]
@mithro I noticed that CI has failed, but is it related to my changes? Is it possible to start it again?

**Comment by mithro** [comment]
@mateusz-holenko Just push an update

**Comment by mateusz-holenko** [comment]
There is a problem with getting `zephyr_sdk` (~ 770MB). The connection speed is low, so it takes too long to download it and the connection is dropped in the meantime.

Is there a simple way to embedd this into the image so it does not have to be downloaded every time?

**Comment by mithro** [comment]
@mateusz-holenko I think we will have to mirror it somewhere that is faster to download from. 770mb should take a couple of seconds...

File: scripts/download-env.sh [hunk 1]
```diff
@@ -486,6 +486,14 @@ if [ "$FIRMWARE" = "zephyr" ]; then
 	echo "Installing pykwalify (pyhton module)"
 	pip install pykwalify
 	check_import pykwalify.core
+
+	# cmake for building Zephyr
+	# Zephyr requires cmake 3.13.1 or higher
+	# installing it via pip provides 3.14.4
+	# at the time of writing this comment
+	echo
+	echo "Installing cmake (pyhton module)"
```
Comment (mithro) [inline scripts/download-env.sh]:
```suggestion
	echo "Installing cmake (python module)"
```

File: scripts/enter-env.sh [hunk 2]
```diff
@@ -344,6 +344,34 @@ check_version ${CPU_ARCH}-elf-gcc $GCC_VERSION || return 1
 #
 #check_version ${CPU_ARCH}-elf-gdb $GDB_VERSION
 
+# Cmake Version
+###############
+if [ "$FIRMWARE" = "zephyr" ]; then
+	set -x
+	# Zephyr requires cmake 3.13.1 or higher
+	CMAKE_VERSION=$(cmake --version | head -n1 | cut -d' ' -f3)
```
Comment (mithro) [inline scripts/enter-env.sh]: There is a check version function?

**Comment by mateusz-holenko** [comment]
The proper place for hosting Zephyr SDK needs to be decided before merging this one.

The current one is for testing purposes.

File: scripts/enter-env.sh (hunk 2 above)
Comment (mateusz-holenko) [reply scripts/enter-env.sh]: `check_version` function verifies an exact version - here I check for version 3.13.1 **or higher**

I agree that it adds complexity, so the other solution I see is to force a 3.14.4 version in pip and simply check here for this version using `check_version`. What do you think @mithro?

Comment (mateusz-holenko) [reply scripts/enter-env.sh]: I fixed the cmake version to 3.14.4

**Comment by mateusz-holenko** [comment]
@mithro Travis CI is constantly failing on installing python using conda. I had the same problems yesterday on my local machine, but today it works fine.

I push-forced the branch today few times, but with no success. Do you have an idea how to solve this issue?

**Comment by mithro** [comment]
Looks like something is screwed up at conda's end?

**Comment by mithro** [comment]
Could also have something to do with our package pins?

**Comment by mithro** [comment]
Blocked on https://github.com/conda/conda/issues/8843

File: scripts/download-env.sh [hunk 3]
```diff
@@ -482,6 +482,16 @@ if [ "$FIRMWARE" = "zephyr" ]; then
 	conda install -y $CONDA_FLAGS pyyaml
 	check_import yaml
 
+	# gperf for Zephyr SDK
```
Comment (mithro) [inline scripts/download-env.sh]: These should be check in `enter-env.sh`...

**Pull request merged**